src/applications/app.helpers.js

```diff
@@ -74,3 +74,111 @@ export function isParcel(appOrParcel) {
 export function objectType(appOrParcel) {
   return isParcel(appOrParcel) ? "parcel" : "application";
 }
+
+export function validateArgumentsTypeAPI(
+  appName,
+  applicationOrLoadingFn,
+  activityFn,
+  customProps
+) {
+  if (typeof appName !== "string" || appName.length === 0)
+    throw Error(
+      formatErrorMessage(
+        20,
+        __DEV__ &&
+          `The first argument to registerApplication must be a non-empty string 'appName'`
+      )
+    );
+
+  if (!applicationOrLoadingFn)
+    throw Error(
+      formatErrorMessage(
+        23,
+        __DEV__ && "The application or loading function is required"
+      )
+    );
+
+  if (typeof activityFn !== "function")
+    throw Error(
+      formatErrorMessage(
+        24,
+        __DEV__ && `The activityFunction argument must be a function`
+      )
+    );
+
+  if (
+    !!customProps &&
+    (typeof customProps !== "object" || Array.isArray(customProps))
+  )
+    throw Error(
+      formatErrorMessage(
+        22,
+        __DEV__ && "optional customProps must be an object"
+      )
+    );
+}
+
+export function validateRegisterObjectAPI(configurationObject) {
+  const validKeys = ["name", "load", "isActive", "customProps"];
+
+  const invalidPassedKeys = Object.keys(configurationObject).reduce(
+    (invalidKeys, prop) =>
+      validKeys.includes(prop) ? invalidKeys : invalidKeys.concat(prop),
+    []
+  );
+
+  if (invalidPassedKeys.length !== 0)
+    throw Error(
+      formatErrorMessage(
+        30,
```

Comment:
I peeked [here](https://github.com/single-spa/single-spa.js.org/tree/f24a828c617ac78779dc11a2ac11d471a62f607f/website/src/components/Errors/Codes), and 30 was  the next valid errorCode